Assets/Scripts/Audio/SFXManager.cs

```diff
@@ -0,0 +1,186 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Audio;
+
+/// <summary>
+/// A useful utility for playing audio that solves the issues of having audiosources in your gameobjects:
+/// + Removes clutter from the inspector of your gameobjects
+/// + Ensures that there are enough audio sources to play audio clips uninterrupted
+/// + Manages the parameters of your audio source components for you (pitch, volume, mixer, clip, etc)
+/// 
+/// Warning: this class is inteded for OneShot sound playing. If looping is inteded you'll need to use a different way.
+/// </summary>
+public class SFXManager : Singleton<SFXManager>
+{
+
+    [Header("Parameters")]
+    [Tooltip("The default sound channel"), SceneEditOnly ]
+    public AudioMixerGroup mixerGroup;
+
+    [SerializeField, Tooltip("How many audio sources to generate"), SceneEditOnly]
+    protected int sourceAmount = 32;
+
+    // Autogenerated
+    protected AudioSource[] sources;
+
+
+#if UNITY_EDITOR
+
+    [SerializeField]
+    protected bool debug = false;
+    protected string dbgClip;
+    protected string dbgPitch;
+    protected string dbgSource;
+
+#endif
+
+    // Start is called before the first frame update
+    void Start()
+    {
+
+        // Create a bunch of audio sources
+        sources = new AudioSource[sourceAmount];
+        for (int i = 0; i < sourceAmount; i++)
+        {
+            // Create a child game object so we don't bloat our inspector
+            GameObject temp = new GameObject("AudioSource ("+i+")");
+            temp.transform.parent = transform;
+            temp.transform.localPosition = Vector3.zero;
+
+            // Add an audiosource on that child gameobject
+            AudioSource source = temp.AddComponent<AudioSource>();
+            source.loop = false;
+            source.playOnAwake = false;
+
+            // Keep track of these audiosource in an array
+            sources[i] = source;
+        }
+    }
+
+    /// <summary>
+    /// Helper function that finds the next available audio source
+    /// </summary>
+    /// <returns></returns>
+    protected AudioSource FindNextSource()
+    {
+        AudioSource source = sources[0];
+        for (int i = 0; i < sources.Length; i++)
+        {
+            if (sources[i].isPlaying == false)
+            {
+                source = sources[i];
+                break;
+            }
+        }
+        return source;
```

Comment:
This will return sources[0] if all sources are preoccupied.

Reply:
Just a minor setback ^-^' ...